api/lib/opentelemetry/context/propagation/propagator.rb

```diff
@@ -46,14 +46,14 @@ def inject(carrier, context = Context.current, &setter)
         # @param [Object] carrier The carrier to extract context from
         # @param [optional Context] context Context to be updated with the state
         #   extracted from the carrier. Defaults to +Context.current+
-        # @param [optional Callable] getter An optional callable that takes a carrier and a key and
-        #   returns the value associated with the key. If omitted the default getter will be used
-        #   which expects the carrier to respond to [] and []=.
+        # @param [optional Getter] getter If the optional getter is provided, it
+        #   will be used to read the header from the carrier, otherwise the default
+        #   getter will be used.
         #
         # @return [Context] a new context updated with state extracted from the
         #   carrier
-        def extract(carrier, context = Context.current, &getter)
-          @extractor.extract(carrier, context, &getter)
+        def extract(carrier, context = Context.current, getter = Context::Propagation.text_map_getter)
```

Comment:
Since we're changing the signature here, can we switch to keyword arguments for the optional args? This will simplify things for users of the Rack getter:
```ruby
    context = OpenTelemetry.propagation.extract(env, getter: OpenTelemetry::Context::Propagation.rack_env_getter)
# vs
    context = OpenTelemetry.propagation.extract(
       env,
       OpenTelemetry::Context.current,
       OpenTelemetry::Context::Propagation.rack_env_getter
     )
```
```suggestion
        def extract(carrier, context: Context.current, getter: Context::Propagation.text_map_getter)
```
If we do this, we should make the same change to `inject`.